tests/autolinker-url.spec.ts

```diff
@@ -1032,6 +1032,13 @@ describe('Autolinker Url Matching >', () => {
         });
     });
 
+    describe('unicode exploits', () => {
+        fit('text with directional change characters should not be linked', () => {
+            expect(autolinker.link('foo.com\u202Ebar.com')).toBe('foo.com\u202Ebar.com');
+            expect(autolinker.link('foo.com\u202Emoc.rab')).toBe('foo.com\u202Emoc.rab');
```

Comment:
In addition to the domain name tests here, we should add tests for when the `\u202E` char appears in the path, query string, and hash sections of the URL.

Also, can we add tests for each of the direction override chars like you had in https://github.com/gregjacobs/Autolinker.js/pull/386?